test/API.spec.ts

```diff
@@ -1,19 +1,35 @@
 import FreeSound from "../index";
+import mockAnalysis from "./__mocks__/sound-analysis.json";
+import mockCombinedSearchResult from "./__mocks__/combined-search.json";
+import mockContentSearchResult from "./__mocks__/content-search.json";
+import mockPack from "./__mocks__/pack.json";
+import mockPackSounds from "./__mocks__/pack-sounds.json";
+import mockSimilarSounds from "./__mocks__/sound-similar.json";
+import mockSound from "./__mocks__/sound.json";
+import mockSoundComments from "./__mocks__/sound-comments.json";
+import mockTextSearch from "./__mocks__/text-search.json";
+import mockUser from "./__mocks__/user.json";
+import mockUserBookCat from "./__mocks__/user-book-cat.json";
+import mockUserBookCatSounds from "./__mocks__/user-book-cat-sounds.json";
+import mockUserPacks from "./__mocks__/user-packs.json";
+import mockUserSounds from "./__mocks__/user-sounds.json";
+import nodeFetch from 'node-fetch';
 
 require("dotenv").config();
 
-function removeVariableProperties(obj: any) {
-  /* eslint-disable */
-  delete obj.num_downloads;
-  delete obj.count;
-  delete obj.results;
-  delete obj.bitrate;
-  delete obj.id;
-  delete obj.average_rating;
-  delete obj.duration;
-  delete obj.ac_analysis;
-  /* eslint-enable */
-  return obj;
+// Import the actual Response object from node-fetch, rather than a mocked version.
+// See https://jestjs.io/docs/bypassing-module-mocks for more information.
+const {Response} = jest.requireActual('node-fetch');
+
+// Mock node-fetch so we can call it without actually making a network request.
+jest.mock('node-fetch');
+
+// Add type information to mocked nodeFetch function.
+// See https://jestjs.io/docs/mock-function-api#typescript for more information.
+const mockNodeFetch = nodeFetch as jest.MockedFunction<typeof nodeFetch>;
+
+const mockResponse = (returnVal: object): void => {
```

Comment:
nit: better name for param would probably be `response` than `returnVal`

Reply:
Note that the ts defs for this fn can be improved by making it generic:
```ts
const mockResponse = <T extends Object>(returnVal: T): void => {
};

const myMockVal = {...}

// ts will infer generic `T` param, no need to explicitly add
mockResponse(myMockVal)
```

Relavent reading:
https://basarat.gitbook.io/typescript/type-system/generics#design-pattern-convenience-generic

Reply:
> nit: better name for param would probably be `response` than `returnVal`

I agree. 7558415 addresses this.

Reply:
> Note that the ts defs for this fn can be improved by making it generic:
> 
> ```ts
> const mockResponse = <T extends Object>(returnVal: T): void => {
> };
> 
> const myMockVal = {...}
> 
> // ts will infer generic `T` param, no need to explicitly add
> mockResponse(myMockVal)
> ```
> 
> Relavent reading:
> https://basarat.gitbook.io/typescript/type-system/generics#design-pattern-convenience-generic

Oh, neat!

Using `const mockResponse = (response: object): void => {`

![Screen Shot 2021-04-28 at 10 11 30 AM](https://user-images.githubusercontent.com/43425812/116444919-217e5980-a80a-11eb-9750-773845366342.png)

Using `const mockResponse = <T extends Object>(response: T): void => {`

![Screen Shot 2021-04-28 at 10 12 12 AM](https://user-images.githubusercontent.com/43425812/116445017-3d81fb00-a80a-11eb-8206-5fcab8839847.png)

Thank you for the link. I was just reading through some of it now. I think b04d689 addresses this, making this pull request ready for another review.